internal/reflect/func.go

```diff
@@ -87,7 +92,9 @@ func WrapFunc(f any) (*Func, error) {
 
 // Call calls the original function with the given arguments.
 func (f *Func) Call(args ...any) ([]Value, error) {
-	if len(args) != len(f.Args) {
+	// TODO: clean up this mess from variadic function arg checking.
+	// also in internal/depinject/container.go
+	if len(args) != len(f.Args) || (f.IsVariadic && len(args) != len(f.Args)-1) {
```

Comment:
Needs to be changed to
```suggestion
	if len(args) != len(f.Args) && !(f.IsVariadic && len(args) == len(f.Args)-1) {
```
or something logically similar